src/common/utxobased/engine/makeUtxoEngineState.ts

```diff
@@ -1115,42 +1023,42 @@ interface ProcessUtxoTransactionArgs extends CommonArgs {
 const processUtxoTransactions = async (
   args: ProcessUtxoTransactionArgs
 ): Promise<void> => {
-  const { scriptPubkey, utxos, currencyInfo, processor, emitter, log } = args
+  const { utxos, processor } = args
 
-  const currentUtxos = await processor.fetchUtxosByScriptPubkey(scriptPubkey)
-  let oldBalance = '0'
-  for (const utxo of currentUtxos) {
-    oldBalance = bs.add(utxo.value, oldBalance)
-    await processor.removeUtxo(utxo.id)
-  }
+  // const currentUtxos = await processor.fetchUtxosByScriptPubkey(scriptPubkey)
+  // let oldBalance = '0'
+  // for (const utxo of currentUtxos) {
+  // oldBalance = bs.add(utxo.value, oldBalance)
+  // await processor.removeUtxo(utxo.id)
+  // }
 
-  let newBalance = '0'
+  // let newBalance = '0'
   for (const utxo of Array.from(utxos)) {
-    newBalance = bs.add(utxo.value, newBalance)
+    // newBalance = bs.add(utxo.value, newBalance)
     await processor.saveUtxo(utxo)
   }
 
-  const diff = bs.sub(newBalance, oldBalance)
-  if (diff !== '0') {
-    log('balance changed:', { scriptPubkey, diff })
-    emitter.emit(
-      EngineEvent.ADDRESS_BALANCE_CHANGED,
-      currencyInfo.currencyCode,
-      diff
-    )
-
-    await processor.updateAddressByScriptPubkey({
-      scriptPubkey,
-      data: {
-        balance: newBalance,
-        used: true
-      }
-    })
-  }
-  setLookAhead({ ...args, ...args.path }).catch(err => {
-    log.error(err)
-    throw err
-  })
+  // const diff = bs.sub(newBalance, oldBalance)
+  // if (diff !== '0') {
+  // log('balance changed:', { scriptPubkey, diff })
+  // emitter.emit(
+  // EngineEvent.ADDRESS_BALANCE_CHANGED,
+  // currencyInfo.currencyCode,
+  // diff
+  // )
+  //
+  // await processor.updateAddressByScriptPubkey({
+  // scriptPubkey,
+  // data: {
+  // balance: newBalance,
+  // used: true
+  // }
+  // })
+  // }
+  // setLookAhead({ ...args, ...args.path }).catch(err => {
+  // log.error(err)
+  // throw err
+  // })
```

Comment:
Cleanup or WIP?

Reply:
WIP - lets discuss and write our conclusion here.

Reply:
Conclusion: We need to keep the workflow the same; this means we also need `fetchUtxos` to support getting by `scriptPubKey`